### ISSUE TITLE
appender.capacity() reports negative values when restarting indexed queues

### ISSUE THREAD

**Issue by chrp3** [OpenHFT/Chronicle-Queue#174]
The following test fails with this exception -ea VM parameter on, but passes without. However appender.capacity() reports negative values.

```
java.lang.IndexOutOfBoundsException: offset out of bounds: 0, len: 4, capacity: -65536
	at net.openhft.lang.io.NativeBytes.actualOffsetChecks(NativeBytes.java:725)
```

Run with version 3.4.4 on windows 8.1, 64-bit Java 1.8.0_45
```java
    @Test
    public void testError() throws Exception {
        String dir = System.getProperty("java.io.tmpdir");
        if (new File(dir, "error.data").exists()) {
            new File(dir, "error.data").delete();
            new File(dir, "error.index").delete();
        }

        write(dir, 2);
        write(dir, 2);
        assertEquals(4, readCount(dir));
    }

    private void write(String dir, int count) throws IOException {
        try (Chronicle chronicle = chronicle(dir)) {
            ExcerptAppender appender = chronicle.createAppender();
            for (int i = 0; i < count; i++) {
                appender.startExcerpt(32);
                appender.writeInt(0, i);
                appender.position(32);
                appender.finish();
            }
            appender.close();
        }
    }

    private int readCount(String dir) throws IOException {
        try (Chronicle chronicle = chronicle(dir)) {
            ExcerptTailer tailer = chronicle.createTailer();
            int i = 0;
            while (tailer.nextIndex()) {
                System.out.println(tailer.readInt());
                i++;
            }
            tailer.close();
            return i;
        }
    }

    private Chronicle chronicle(String dir) throws IOException {
        return ChronicleQueueBuilder
                .indexed(dir, "error")
                .dataBlockSize(64)
                .indexBlockSize(64)
                .build();
    }
```

**Comment by peter-lawrey** [comment]
These block sizes shouldn't be allowed. The minimum page size is 4 KB on
most machines and 64 KB on some versions of windows.
Can you try block sizes of 64 KB to see if this still produces an error?

Thank you,
Peter.

**Comment by chrp3** [comment]
Changed the number of writes and the length of each message:
```java
        write(dir, 1024 * 1024 * 5);
        write(dir, 1024 * 1024 * 5);
        assertEquals(1024 * 1024 * 5 * 2, readCount(dir));

       // ----
                appender.startExcerpt(64);
                appender.writeInt(0, i);
                appender.position(64);
                appender.finish();
```

Results with different data block sizes**:
```
dataBlockSize(64 * 1024) # offset out of bounds: 0, len: 4, capacity: -25690112
dataBlockSize(128 * 1024) # offset out of bounds: 0, len: 4, capacity: -25690112
dataBlockSize(32 * 1024 * 1024) # offset out of bounds: 0, len: 4, capacity: 0
dataBlockSize(64 * 1024 * 1024) # offset out of bounds: 0, len: 4, capacity: 0
dataBlockSize(128 * 1024 * 1024) # offset out of bounds: 0, len: 4, capacity: -67108864
dataBlockSize(256 * 1024 * 1024) # offset out of bounds: 0, len: 4, capacity: -67108864
```
** indexBlockSize is not set.

**Comment by lburgazzoli** [comment]
@chrp3 I've implemented a fix in the latest 3.4.5-SNAPSHOT, could you give it a try ?

**Comment by chrp3** [comment]
I'm unable to reproduce the error after the fix in 3.4.5-SNAPSHOT.

Thanks for the swift resolution!

Regards,
Christian
